packages/wagmi/src/client.ts

```diff
@@ -754,6 +754,14 @@ export class Web3Modal extends Web3ModalScaffold {
             }
             break
           }
+          case W3mFrameConstants.RPC_RESPONSE_TYPE_OBJECT: {
+            if (super.isTransactionStackEmpty()) {
+              super.close()
+            } else {
+              super.popTransactionStack()
+            }
+            break
+          }
```

Comment:
@enesozturk @svenvoskamp any issue with this? will this affect swap / send or other flows?

Reply:
@chris13524 this might be causing the issue "After switching networks it closes the modal as opposed to showing the switch button again?"

Reply:
@tomiir I think this is causing the few test to fails, I tried removing this particular change and then run test (eventually all the test passed), but the issue of modal not getting close was introduced.
Please let me know if there is any other way to handle this change.